src/Hosting/Hosting/src/Internal/HostingLoggerExtensions.cs

```diff
@@ -132,7 +132,7 @@ public int Count
             public HostingLogScope(HttpContext httpContext, string correlationId)
             {
                 _traceIdentifier = httpContext.TraceIdentifier;
-                _path = httpContext.Request.Path.ToString();
+                _path = (httpContext.Request.PathBase+httpContext.Request.Path).ToString();
```

Comment:
This always ends up allocating a string. I suggest checking for an empty `PathBase`:

```
_path = (httpContext.Request.PathBase.HasValue ? httpContext.Request.PathBase + httpContext.Request.Path : httpContext.Request.Path).ToString();
```

The majority of request do not have a PathBase.